extensions/websockets-next/deployment/src/test/java/io/quarkus/websockets/next/test/telemetry/AbstractWebSocketsOnMessageTest.java

```diff
@@ -0,0 +1,349 @@
+                .withApplicationRoot(root -> root
+                        .addPackage(endpointsPackage)
+                        .addClasses(WSClient.class, Connection.class, MetricsAsserter.class,
+                                AbstractWebSocketsOnMessageTest.class)
+                        .addAsResource(new StringAsset("""
+                                bounce-endpoint.prefix-responses=true
+                                """), "application.properties"))
+                .setForcedDependencies(
+                        List.of(Dependency.of("io.quarkus", "quarkus-micrometer-registry-prometheus-deployment",
+                                Version.getVersion())));
+    }
+
+    protected final MetricsAsserter asserter = new MetricsAsserter();
+
+    @TestHTTPResource("bounce")
+    URI bounceUri;
+
+    @TestHTTPResource("/")
+    URI baseUri;
+
+    @TestHTTPResource("received-single-text-response-none")
+    URI singleTextReceived_NoResponse_Uri;
+
+    @TestHTTPResource("received-single-text-response-multi-text")
+    URI singleTextReceived_MultiTextResponse_Uri;
+
+    @TestHTTPResource("received-multi-text-response-none")
+    URI multiTextReceived_NoResponse_Uri;
+
+    @TestHTTPResource("received-multi-text-response-single-text")
+    URI multiTextReceived_SingleTextResponse_Uri;
+
+    @TestHTTPResource("received-multi-text-response-multi-text")
+    URI multiTextReceived_MultiTextResponse_Uri;
+
+    @TestHTTPResource("received-single-text-response-uni-text")
+    URI singleTextReceived_UniTextResponse_Uri;
+
+    @TestHTTPResource("received-single-dto-response-single-dto")
+    URI singleDtoReceived_SingleDtoResponse_Uri;
+
+    @TestHTTPResource("received-single-dto-response-none")
+    URI singleDtoReceived_NoResponse_Uri;
+
+    @TestHTTPResource("received-single-dto-response-uni-dto")
+    URI singleDtoReceived_UniDtoResponse_Uri;
+
+    @TestHTTPResource("received-single-dto-response-multi-dto")
+    URI singleDtoReceived_MultiDtoResponse_Uri;
+
+    @TestHTTPResource("received-multi-dto-response-none")
+    URI multiDtoReceived_NoResponse_Uri;
+
+    @TestHTTPResource("received-multi-dto-response-single-dto")
+    URI multiDtoReceived_SingleDtoResponse_Uri;
+
+    @TestHTTPResource("received-multi-dto-response-multi-dto")
+    URI multiDtoReceived_MultiDtoResponse_Uri;
+
+    @TestHTTPResource("broadcast")
+    URI broadcast_Uri;
+
+    @Inject
+    Vertx vertx;
+
+    protected abstract boolean binaryMode();
+
+    protected abstract WebSocketConnector<?> bounceClientConnector();
+
+    protected abstract WebSocketConnector<?> multiClientConnector();
+
+    @Test
+    public void testServerEndpoint_SingleTextReceived_NoSent() {
+        // endpoint: void onMessage(String message)
+        var connection = Connection.of(singleTextReceived_NoResponse_Uri, false, 0, binaryMode(), "Ballad of a Prodigal Son");
+        sendAndAssertResponses(vertx, connection);
+        asserter.assertMetrics(0, 1, connection);
```

Comment:
I would prefer explicit assertions to really understand what to expect here... Should just count the nr of metrics be ok?... I would prefer an explicit declaration including the assertion of the path.

Reply:
On the other hand, if it is a parametrized test, then explicit assertions is more than bearable.

Reply:
In parametrized tests, I have added "explicit" (as I understand it, feel free to correct me) assertions as well as metrics assertions per path

Reply:
Looks much better, thanks!